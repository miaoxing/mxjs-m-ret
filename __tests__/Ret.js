import Ret from '..';
import {render} from '@testing-library/react';
import {Ret as RetObject} from 'miaoxing';

describe('Ret', () => {
  test('basic', () => {
    const result = render(<Ret ret={RetObject.err('error')}/>);
    expect(result.container).toMatchSnapshot();
  });

  test('empty ret show loading', () => {
    const {queryByText} = render(<Ret/>);
    expect(queryByText('加载中...')).not.toBeNull();
  });

  test('empty ret object show loading', () => {
    const {queryByText} = render(<Ret ret={{}}/>);
    expect(queryByText('加载中...')).not.toBeNull();
  });
});
