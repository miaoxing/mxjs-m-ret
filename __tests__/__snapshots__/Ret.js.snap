// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ret basic 1`] = `
<div>
  <taro-block-core>
    <taro-view-core
      style="display: flex; height: 100vh; text-align: center; align-items: center; justify-content: center; flex-direction: column;"
    >
      <taro-image-core
        mode="center"
        src="https://gw.alipayobjects.com/zos/rmsportal/GIyMDJnuqmcqPLpHCSkj.svg"
        style="margin-bottom: 1rem;"
      />
      <taro-view-core
        style="margin-bottom: 1rem; font-size: 1.125rem;"
      >
        error
      </taro-view-core>
    </taro-view-core>
  </taro-block-core>
</div>
`;
